orttraining/orttraining/python/training/ortmodule/_execution_agent.py

```diff
@@ -11,6 +11,8 @@
 
 class ExecutionAgentOutput(object):
     def __init__(self, ortvalues, run_id=None):
+        if isinstance(ortvalues, list):
+            raise TypeError("ortvalues must be of type 'OrtValueVector'.")
```

Comment:
Is there a way to do something similar to `InferenceSession` or whatever is used for inference-only? to make sure `VectorORTValue` works for them too after this change

Reply:
I added more test in class InferenceSession (python class, not pybind11). Users should not be impacted by the changes as the frontend wraps the class pybind11 OrtValue in a python class with the same name.